### PR TITLE
Fix doc generator

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR addressed errors raised when running `scripts/generate_api_docs.py`

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by angrybayblade — COMMENTED** [review]
Turns out most of the errors were caused by badly formatted doc strings, I've fixed them but there's formatting issues that still remains which is

https://github.com/valory-xyz/open-aea/blob/f2c207fe357e463039550c52014fa9ed7433fd8d/aea/protocols/dialogue/base.py#L244

here, I reformatted it but it still causes error.

**Review by dvilelaf — DISMISSED** [review]

**Comment by angrybayblade** [comment]
> Turns out most of the errors were caused by badly formatted doc strings, I've fixed them but there's formatting issues that still remains which is
> 
> https://github.com/valory-xyz/open-aea/blob/f2c207fe357e463039550c52014fa9ed7433fd8d/aea/protocols/dialogue/base.py#L244
> 
> here, I reformatted it but it still causes error.

So I tried fixing this but It seems the parser formats the code before rendering the markdown so we'll have to keep this as it is

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
I locally still get failures:

```
[WARNING - pydoc_markdown.contrib.processors.crossref]: 1 cross-reference(s) could not be resolved:
  aea.configurations.data_types.Dependencies: name
```

```
Processing aea/protocols/dialogue/__init__.py... Skipping, it's in ignore patterns
Traceback (most recent call last):
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/yapf/yapflib/pytree_utils.py", line 119, in ParseCodeToTree
    tree = parser_driver.parse_string(code, debug=False)
  File "/Users/davidminarsch/.pyenv/versions/3.10.4/lib/python3.10/lib2to3/pgen2/driver.py", line 103, in parse_string
    return self.parse_tokens(tokens, debug)
  File "/Users/davidminarsch/.pyenv/versions/3.10.4/lib/python3.10/lib2to3/pgen2/driver.py", line 71, in parse_tokens
    if p.addtoken(type, value, (prefix, start)):
  File "/Users/davidminarsch/.pyenv/versions/3.10.4/lib/python3.10/lib2to3/pgen2/parse.py", line 162, in addtoken
    raise ParseError("bad input", type, value, context)
lib2to3.pgen2.parse.ParseError: bad input: type=12, value=',', context=('', (1, 15))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/yapf/yapflib/pytree_utils.py", line 125, in ParseCodeToTree
    tree = parser_driver.parse_string(code, debug=False)
  File "/Users/davidminarsch/.pyenv/versions/3.10.4/lib/python3.10/lib2to3/pgen2/driver.py", line 103, in parse_string
    return self.parse_tokens(tokens, debug)
  File "/Users/davidminarsch/.pyenv/versions/3.10.4/lib/python3.10/lib2to3/pgen2/driver.py", line 71, in parse_tokens
    if p.addtoken(type, value, (prefix, start)):
  File "/Users/davidminarsch/.pyenv/versions/3.10.4/lib/python3.10/lib2to3/pgen2/parse.py", line 162, in addtoken
    raise ParseError("bad input", type, value, context)
lib2to3.pgen2.parse.ParseError: bad input: type=12, value=',', context=('', (1, 15))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/yapf/yapflib/yapf_api.py", line 183, in FormatCode
    tree = pytree_utils.ParseCodeToTree(unformatted_source)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/yapf/yapflib/pytree_utils.py", line 131, in ParseCodeToTree
    raise e
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/yapf/yapflib/pytree_utils.py", line 129, in ParseCodeToTree
    ast.parse(code)
  File "/Users/davidminarsch/.pyenv/versions/3.10.4/lib/python3.10/ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 1
    class Dialogue(, metaclass=_DialogueMeta): pass
                  ^
SyntaxError: expected ':'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/bin/pydoc-markdown", line 8, in <module>
    sys.exit(cli())
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/click/core.py", line 1126, in __call__
    return self.main(*args, **kwargs)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/click/core.py", line 1051, in main
    rv = self.invoke(ctx)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/click/core.py", line 1393, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/click/core.py", line 752, in invoke
    return __callback(*args, **kwargs)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/main.py", line 344, in cli
    session.render(pydocmd)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/main.py", line 138, in render
    config.render(modules)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/__init__.py", line 179, in render
    self.renderer.render(modules)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/contrib/renderers/markdown.py", line 461, in render
    self._render_to_stream(modules, sys.stdout)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/contrib/renderers/markdown.py", line 423, in _render_to_stream
    return self.render_single_page(stream, modules)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/contrib/renderers/markdown.py", line 443, in render_single_page
    self._render_recursive(fp, 1, m)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/contrib/renderers/markdown.py", line 368, in _render_recursive
    self._render_recursive(fp, level, member)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/contrib/renderers/markdown.py", line 365, in _render_recursive
    self._render_object(fp, level, obj)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/contrib/renderers/markdown.py", line 350, in _render_object
    self._render_signature_block(fp, obj)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/contrib/renderers/markdown.py", line 327, in _render_signature_block
    code = self._format_classdef_signature(obj)
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/contrib/renderers/markdown.py", line 312, in _format_classdef_signature
    code = self._yapf_code(code + ': pass').rpartition(':')[0].strip()
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/pydoc_markdown/contrib/renderers/markdown.py", line 281, in _yapf_code
    return FormatCode(code, style_config=style)[0]
  File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/yapf/yapflib/yapf_api.py", line 186, in FormatCode
    raise errors.YapfError(errors.FormatErrorMsg(e))
yapf.yapflib.errors.YapfError: <unknown>:1:15: expected ':'
```

Also, why do we not throw a proper exception? Now I need to manually parse through the log output...

**Comment by angrybayblade** [comment]
> I locally still get failures:
> 
> ```
> [WARNING - pydoc_markdown.contrib.processors.crossref]: 1 cross-reference(s) could not be resolved:
>   aea.configurations.data_types.Dependencies: name
> ```
>  ...
> ```
> Processing aea/protocols/dialogue/__init__.py... Skipping, it's in ignore patterns
> Traceback (most recent call last):
>   File "/Users/davidminarsch/.local/share/virtualenvs/open-aea-WmhCcIRs/lib/python3.10/site-packages/yapf/yapflib/pytree_utils.py", line 119, in ParseCodeToTree
>     tree = parser_driver.parse_string(code, debug=False)
>   File "/Users/davidminarsch/.pyenv/versions/3.10.4/lib/python3.10/lib2to3/pgen2/driver.py", line 103, in parse_string
>     return self.parse_tokens(tokens, debug)
>   File "/Users/davidminarsch/.pyenv/versions/3.10.4/lib/python3.10/lib2to3/pgen2/driver.py", line 71, in parse_tokens
>     if p.addtoken(type, value, (prefix, start)):

> Also, why do we not throw a proper exception? Now I need to manually parse through the log output...

The local failure is caused by this https://github.com/valory-xyz/open-aea/pull/315#pullrequestreview-1107834403

We don't throw proper exception because we run the pydoc in a subprocess and don't check for the exit code also stderr is also not set the it'll write the error to console and since we don't check for exit code we can't raise an exception if the process fails

File: scripts/generate_api_docs.py [hunk 1]
```diff
@@ -52,6 +52,7 @@
     Path("aea", "decision_maker", "scaffold.py"),
     Path("aea", "error_handler", "scaffold.py"),
     Path("aea", "test_tools", "click_testing.py"),
+    Path("aea", "protocols", "dialogue", "base.py"),
```
Comment (DavidMinarsch) [inline scripts/generate_api_docs.py:55]: That's not what I mean - I meant ignoring specific errors so we can keep the file

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**